### PR TITLE
chore: clean up ds-test failure related code

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: crates/evm/core/src/backend/mod.rs [hunk 1]
```diff
@@ -583,80 +578,12 @@ impl Backend {
...
-    /// Instead, it stores whether an `assert` failed in a boolean variable that we can read
-    pub fn is_failed_test_contract(&self, address: Address) -> bool {
-        /*
-        contract DSTest {
-            bool public IS_TEST = true;
-            // slot 0 offset 1 => second byte of slot0
-            bool private _failed;
-        }
-        */
-        let value = self.storage_ref(address, U256::ZERO).unwrap_or_default();
```
Comment (DaniPopes) [inline crates/evm/core/src/backend/mod.rs:608]: 1. this is wrong
2. the location of the failed variable is not specified and should not be assumed

**Review by DaniPopes — COMMENTED** [review]
I removed the extra checks, they're not worth it without breaking changes. If we do end up breaking this, we should stop calling the `failed` function and just rely on whatever is stored in the global failure slot in the cheatcodes address.

File: crates/evm/evm/src/executors/mod.rs [hunk 2]
```diff
@@ -513,13 +508,34 @@ impl Executor {
...
-            // a failure occurred in a reverted snapshot, which is considered a failed test
             return false;
         }
 
-        let mut success = !reverted;
-        if success {
+        // Check the global failure slot.
+        // TODO: Wire this up
+        let legacy = true;
+        if !legacy {
```
Comment (DaniPopes) [inline crates/evm/evm/src/executors/mod.rs:524]: This will be done in a follow up

**Review by mattsse — APPROVED** [review]
that is much simpler

**Pull request merged**